include/roboptim/core/solver-state.hh

```diff
@@ -82,9 +82,9 @@ namespace roboptim
     typedef P problem_t;
 
     /// \brief Import argument types from problem
-    typedef typename P::function_t::argument_t argument_t;
-    typedef typename P::function_t::argument_ref argument_ref;
-    typedef typename P::function_t::const_argument_ref const_argument_ref;
+    typedef typename P::vector_t argument_t;
+    typedef typename P::vector_t& argument_ref;
+    typedef const typename P::vector_t& const_argument_ref;
```

Comment:
Why was this needed? The function [holds the proper typedefs](https://github.com/roboptim/roboptim-core/blob/master/include/roboptim/core/function.hh#L173).